compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt

```diff
@@ -30,19 +31,20 @@ import androidx.compose.ui.input.pointer.pointerHoverIcon
 import androidx.compose.ui.input.pointer.pointerInput
 import androidx.compose.ui.layout.Measurable
 import androidx.compose.ui.unit.Constraints
-import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
 import java.awt.Cursor
 import java.awt.Dimension
 import java.awt.MouseInfo
 import java.awt.Point
 import java.awt.Window
 
-internal val DefaultBorderThickness = 8.dp
+/**
+ * The thickness of the border resizer for undecorated windows.
+ */
+val LocalUndecoratedWindowResizerBorderThickness = staticCompositionLocalOf { 8.dp }
```

Comment:
> We could expose it, or provide an additional parameter to Window and DialogWindow, but each of those comes with its own considerations and downsides.

This way is more friendly for users. I would also stick to it, even if it is harder to design. If we add a local for every configuration, the API will quickly become hard to understand.

We can add something like this (this is an example, I haven't think about it thoroughly):
```
@ExperimentalComposeUIApi Window(undecoratedFrame = WindowUndecoratedFrame(resizerBorderThickness = borderThickness)
```
(and create an issue to remove experimental in 1.8)

Reply:
> If we add a local for every configuration, the API will quickly become hard to understand.

I don't think it will be hard to understand - what is hard about a simple composition local providing a value?

The downside of a composition local is that it's harder to discover. The downside of an explicit parameter is API bloat, and looking at `Window`, it already has 14 parameters.

Additionally, changing the thickness of the undecorated resizer is rather niche, so I'm not sure it's good to have it as an explicit argument.

Finally, it doesn't work well with the current `undecorated` argument because it's not orthogonal. You can specify `undecorated = false` (or leave it the default) and `undecoratedFrame = ...` and then it's not clear what happens. We could add another overload to deal with that. Something like:
```
@Composable
Window(
    ...
    decoration: WindowDecoration = WindowDecoration.SystemDecoration
)

interface WindowDecoration {
    object SystemDecoration : WindowDecoration
    class Undecorated(val resizerThickness: Dp = DefaultWindowResizerThickness) : WindowDecoration
}
```

I'm not strongly against doing this, but API bloat is something to consider here...

Reply:
Your suggestion with a new overload `WindowDecoration` looks good.